### DISCUSSION TITLE
Trouble Compiling the FundMe.sol file on Zksync

### DISCUSSION THREAD

**Discussion by odibaegba** [Cyfrin/foundry-full-course-cu#2893] [Q&A]
I keep getting this error .

Failed to compile:
Error HH411: The library @chainlink/contracts@1.2.0, imported from contracts/FundMe.sol, is not installed. Try installing it using npm.
HardhatError: HH411: The library @chainlink/contracts@1.2.0, imported from contracts/FundMe.sol, is not installed. Try installing it using npm.
    at Resolver.resolveImport (/opt/app/api/hardhat_env/node_modules/hardhat/src/internal/solidity/resolver.ts:254:15)
    at async /opt/app/api/hardhat_env/node_modules/hardhat/src/internal/solidity/dependencyGraph.ts:188:28
    at async Promise.all (index 0)
    at async DependencyGraph._addDependenciesFrom

**Comment by odibaegba** [comment]
<img width="1440" alt="Screenshot 2024-11-20 at 19 28 11" src="https://github.com/user-attachments/assets/140f111c-97a0-4bdc-8b8c-a92a91b9a0e9">

**Comment by EngrPips** [comment]
Please show the `zksync` plugin page.

**Reply by odibaegba** [reply]
okay

**Comment by odibaegba** [comment]
<img width="1440" alt="Screenshot 2024-11-22 at 11 04 32" src="https://github.com/user-attachments/assets/b0d1671a-7ee0-4730-aefc-4234ad6c3351">

here you go

**Reply by EngrPips** [reply]
Read this discussion https://github.com/Cyfrin/foundry-full-course-cu/discussions/2903 It should help you.

**Reply by odibaegba** [reply]
okay.. thanks.. let me check it out

**Reply by EngrPips** [reply]
Anytime, Friend.

**Reply by odibaegba** [reply]
@EngrPips  so i check this out and noticed that was a deployment issue... mine is a compiling issue though.
the code compiles pefectly on ethSepolia but dose not compile whenever i try compiling on ZKsync

**Reply by EngrPips** [reply]
> <img alt="Screenshot 2024-11-22 at 11 04 32" width="1440" src="https://private-user-images.githubusercontent.com/112698852/388892504-b0d1671a-7ee0-4730-aefc-4234ad6c3351.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIyNzA5NzEsIm5iZiI6MTczMjI3MDY3MSwicGF0aCI6Ii8xMTI2OTg4NTIvMzg4ODkyNTA0LWIwZDE2NzFhLTdlZTAtNDczMC1hZWZjLTQyMzRhZDZjMzM1MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyMlQxMDE3NTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMTM4MTYyMzI3NTZhZmExMTliZjQyZGUyOWQ5MjExYjZkYWQxNGI5Y2FhMTg3NTFhMTFjMTkyMGRiY2FkMWQxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.CEU6K7WyN0FkBoN6psdB29HBXRg5BZUUCMNfuvhRJYQ">
> here you go

This image shows that you didn't select any account, I think you need to select account to deploy.

**Reply by odibaegba** [reply]
okay.. i will try that out on Remix.
thanks for the support mate.

**Reply by EngrPips** [reply]
You are welcome. Also, did you put your solidity file inside the `contracts` folder? And are you working in the `default-workspace`?

**Reply by odibaegba** [reply]
yes, I did.

**Reply by odibaegba** [reply]
I do not understand what the default work space means though

**Reply by odibaegba** [reply]
yes, I am working on the default workespace.. I just checked now

**Reply by EngrPips** [reply]
Oh good, Things should work correctly.

**Comment by crossterra-agent** [comment]
## APPROVED

The revised plan addresses both prior validator concerns and all file references check out against the codebase:

- **S01**: `storefront_v2/src/__tests__/utils/supportMailto.test.ts` confirmed at lines 24-44 with `expectedSlugs` (6 elements, lines 25-32), test name `has an entry for every support-page brand slug (6)` (line 34), and `toHaveLength(6)` (line 35). Plan correctly enumerates all three updates plus a new Crossterra case.
- **S03**: Verified `en.json` line 675 `"support": {` with `"shopnewrepublic": {` at line 845 — the nested shape in the revised example matches the actual file structure (not a dotted top-level key).
- `StorefrontSlug.CROSSTERRA` exists at `storefrontStore.ts:12` (plan correctly notes do not redefine).
- `ABOUT_PAGE_ALLOWED_SLUGS` (lines 35-43), `SUPPORT_PAGE_ALLOWED_SLUGS` (lines 25-33), and `SUPPORT_CLIENT_NAMES` (6 entries, `crossterra` missing) all match.
- Page-client conditional chains (`AboutPageClient.tsx:30-44`, `SupportPageClient.tsx:35-41`) and the `ripndipFaqs.tsx:32-55` link-splitting pattern exist exactly as described.
- `bcbg.tsx`/`steals.tsx` `Record<string, React.ReactNode>` + `getXAbout(locale)` fallback pattern confirmed.
- `ko.json` and `es-MX.json` both already carry `support.ripndip`/`support.shopnewrepublic` blocks, supporting the plan's requirement to add `support.crossterra` to all four messages files.
- Footer (`StorefrontFooter.tsx:82-86`) renders `/about` unconditionally and gates `/support` on `isShowSupport` — matches plan claim that adding the slug to allow-lists is sufficient to surface the Support footer link.

No migrations, auth, error-handling, or feature-flag concerns (static content gated by an existing allow-list). Step ordering (data → translations → UI → verify) respects dependencies; each step is single-session-sized. Ready to implement.

**Reply by EngrPips** [reply]
Is this a solution to the `FundMe.sol` compilation issue on ZKsync?
